### PR TITLE
Query for resource keys.

### PR DESCRIPTION
**Pull request by Deren-Liao**
This pull request adds get resource keys, get resource type values methods.  It also updates the current List log id method with the resource key prefix. 
These are extension methods.

This also update how the view model initialize the resource types and log ids.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs [hunk 1]
```diff
@@ -45,16 +52,62 @@ public LoggingDataSource(string projectId, GoogleCredential credential, string a
...
+                },
+                x => x.ResourceKeys,
+                x => x.NextPageToken);
+        }
+
+        /// <summary>
+        /// List all resource type values for the given resource type and resource key.
+        /// </summary>
+        /// <param name="resourceType">Required, the resource type.</param>
+        /// <param name="resourceKey">Optional, the resource key as prefix.</param>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs]: If this is optional should it default to null?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs [hunk 2]
```diff
@@ -71,13 +124,16 @@ public LoggingDataSource(string projectId, GoogleCredential credential, string a
         /// The size of entire set of log names is small. 
         /// Batch all in one request in unlikely case it spans multiple pages.
         /// </summary>
-        public Task<IList<string>> ListProjectLogNamesAsync()
+        public Task<IList<string>> ListProjectLogNamesAsync(string resourceType, IList<string> resourceKeys)
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs]: Comments for the params?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 3]
```diff
@@ -650,28 +657,36 @@ private string ComposeTextSearchFilter()
...
-            string[] _filterProperties = new string[] {
-                nameof(SelectedLogSeverity),
-                nameof(SelectedResource),
-                nameof(LogIdsList.SelectedLogId),
-            };
-            if (_filterProperties.Contains(e.PropertyName))
+            switch (e.PropertyName)
             {
-                Reload();
+                case nameof(SelectedTimeZone):
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: This change seems a little out of place for this PR.  Should this be here?  If so can you update the description

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs (hunk 1 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs (hunk 2 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs (hunk 3 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: It's related.  I'll update description.

**Comment by Deren-Liao** [comment]
Updated, please review again. Thank you!

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs [hunk 4]
```diff
@@ -124,16 +124,21 @@ public Task<IList<MonitoredResourceDescriptor>> GetResourceDescriptorsAsync()
         /// The size of entire set of log names is small. 
         /// Batch all in one request in unlikely case it spans multiple pages.
         /// </summary>
-        public Task<IList<string>> ListProjectLogNamesAsync(string resourceType, IList<string> resourceKeys)
+        /// <param name="resourceType">The resource type, i.e gce_instance.</param>
+        /// <param name="resourcePrefixList">
+        /// A list of resource prefixes.
+        /// i.e,  for resource type app engine, the prefixe can be the module ids. 
+        /// </param>
+        public Task<IList<string>> ListProjectLogNamesAsync(string resourceType, IList<string> resourcePrefixList)
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs]: resourcePrefixList looks optional?  Can you note that in the comment and add a default?

Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/LoggingDataSource.cs]: Sure. I'll do before merging.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 5]
```diff
@@ -509,7 +512,7 @@ private void SetServerRequestCompleteStatus()
         /// </summary>
         private void Reload()
         {
-            Debug.WriteLine("Entering Reload()");
+            Debug.WriteLine($"Entering Reload(), thread id {Thread.CurrentThread.ManagedThreadId}");
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: Curious, would this actually be ever called **not** on the main thread?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 6]
```diff
@@ -650,28 +657,36 @@ private string ComposeTextSearchFilter()
         /// </summary>
         private async Task PopulateLogIds()
         {
-            IList<string> logIdRequestResult = null;
-            logIdRequestResult = await _dataSource.Value.ListProjectLogNamesAsync();
+            IList<string> logIdRequestResult
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: Nit, there's really no line limit, just _"be reasonable"_ that way we can avoid this Java-isms.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs (hunk 5 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: No, I was debugging it.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs (hunk 6 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: Agree. I fix it in next PR.

**Pull request merged**